addons/point_of_sale/models/res_partner.py

```diff
@@ -38,3 +39,13 @@ def create_from_ui(self, partner):
             partner['lang'] = self.env.user.lang
             partner_id = self.create(partner).id
         return partner_id
+
+    @api.multi
+    def unlink(self):
+        running_sessions = self.env['pos.session'].search([('state', '!=', 'closed')])
+        if running_sessions:
+            raise UserError(
+                _("You cannot delete contacts while there are active PoS sessions. Close the session(s) %s first.")
```

Comment:
This is the only error message in this diff that actually tell the user what to do to solve the problem; please make sure all other error messages do the same